docs/specs/mlnet-cli/MLNET-CLI-Specs.md

```diff
@@ -0,0 +1,969 @@
+- A CLI is aligned to modern cross-platform frameworks experience such as .NET Core, Python, Node, etc. 
+
+## Positioning the CLI and AutoML in ML.NET
+
+![alt text](Images/MLNET-AutoML-Positioning.png "ML.NET and AutoML")
+
+As mentioned at the begining of the spec doc, the CLI will be branded as the ML.NET CLI since this CLI will also have additional features where AutoML is not needed.
+
+The .NET AutoML API will be used for the most important foundational features that will be part of the ML.NET CLI, though, as shown in the following diagram:
+
+![alt text](Images/MLNET-AutoML-Working-Diagram.png "AutoML working inside the ML.NET CLI")
+
+
+## Design criteria
+
+- The ML.NET CLI will create **four main types of results** related to AutoML:
+    - Best trained models (Serialized model files, such as .ZIP files).
+    - C# ML.NET sample code consuming/scoring the models.
+    - C# ML.NET pipeline implementation code for each possible "best model" configuration.
+    - Result report with ML analysis of the models (*Mostly since v1.0*)
+
+- The ML.NET CLI automation will be able to run locally, on any development environment PC (Windows, Mac or Linux).
+- Future versions will be able to run AutoML compute and other compute processes (such as a regular model training) in Azure.
+- The ML.NET CLI  will consume the AutoML API (Microsoft.ML.Auto NuGet package) which will only consume public surface of ML.NET.
+- The CLI proposed here will not provide for “continue sweeping” after sweeping has ended.
+- When running locally with the by default behaviour (no Azure), the CLI will not make any webservice calls and will not require any authentication.
+- The CLI will provide feedback output (such as % work done or high level details on what's happening under the covers) while working on the long-running tasks.
+- The ML.NET CLI will be aligned and integrated to the [.NET Core CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/?tabs=netcore2x). A good approach is to implement the ML.NET CLI as a [.NET Core Global Tool](https://docs.microsoft.com/en-us/dotnet/core/tools/global-tools) (i.e. named "mlnet" package) on top of the "dotnet CLI". 
+  - [System-Command-Line](https://github.com/dotnet/command-line-api) package can precisely help on those typical CLI console apps features (CLI plumbing) so if can be implemented a lot easier than by implementing a CLI app from scratch in features like command line parsing, invocation and rendering of terminal output. Otherwise, a CLI implementation starting from scratch can be significantly more costly to develop.
+
+### CLI default behaviour and overridability
+
+The CLI will have default behavior for each of these mentioned features – however the CLI by default settings should be able to be overridden by providing new/overriden values in the console command (and optionally the advanced configuration .YAML file and response file .rsp placed along with the CLI executable).
```

Comment:
>.YAML file and response file .rsp [](start = 242, length = 33)

one or the other but not both.  Having CLI, YAML and RSP is too much. How about CLI + YAML.